engine/apps/grafana_plugin/views/sync_v2.py

```diff
@@ -25,7 +28,14 @@ class SyncV2View(APIView):
     authentication_classes = (BasePluginAuthentication,)
 
     def do_sync(self, request: Request) -> Organization:
-        serializer = SyncDataSerializer(data=request.data)
+        if request.headers.get("Content-Encoding") == "gzip":
```

Comment:
Trying this route rather than middleware since we only want to deal with compression on here.